query_base.go

```diff
@@ -1352,3 +1354,113 @@ func (ih *idxHintsQuery) bufIndexHint(
 	b = append(b, ")"...)
 	return b, nil
 }
+
+//------------------------------------------------------------------------------
+
+type orderLimitOffsetQuery struct {
+	order []schema.QueryWithArgs
+
+	limit  int32
+	offset int32
+}
+
+func (olfq *orderLimitOffsetQuery) addOrder(orders ...string) {
```

Comment:
Why use olfq instead of q?

Reply:
Good question! Probably inspired by `func (ih *idxHintsQuery)`. Fixed.